### DISCUSSION TITLE
Aqara P1 Sensor Error when trying to adjust Detection Interval or any other setting in zigbee2mqtt

### DISCUSSION THREAD

**Discussion by mdllocalhost** [Koenkk/zigbee2mqtt#12889] [Devices]
Hi guys,

I have been trying to troubleshoot this error after I try to adjust the Detection Interval of the Aqara P1 within the Exposes Tab in zigbee2mqtt:

2022-06-20 19:18:19Publish 'set' 'detection_interval' to 'Bedroom_Aqara_Motion Sensor P1' failed: 'Error: Write 0x54ef441000490838/1 aqaraOpple({"258":{"value":[2],"type":32}}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4447,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'MAC transaction expired' (240))'

I am also unable to adjust any other parameter within the Exposes Tab and I am stuck in the default setting for the sensor.

This is the error I get when I try to change the Motion Sensitivity:

2022-06-20 19:24:31Publish 'set' 'motion_sensitivity' to 'Bedroom_Aqara_Motion Sensor P1' failed: 'Error: Write 0x54ef441000490838/1 aqaraOpple({"268":{"value":3,"type":32}}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4447,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'MAC transaction expired' (240))'

I would truly appreciate any help that you could provide!

**Comment by Koenkk** [comment]
Make sure to wakeup the sensor right before setting the value.

**Reply by mdllocalhost** [reply]
Will do thank you Koenkk!!!

**Reply by mdllocalhost** [reply]
> Make sure to wakeup the sensor right before setting the value.

Hi Koenkk, I am experiencing the same issue even after awakening the sensor, any suggestions? 

Here are the logs after I awake the sensor:

Info 2022-06-26 09:56:40MQTT publish: topic 'zigbee2mqtt/Hallway_Aqara_Motion Sensor P1', payload '{"battery":100,"detection_interval":30,"device_temperature":21,"illuminance":986,"last_seen":"2022-06-26T09:56:39-04:00","linkquality":76,"motion_sensitivity":"medium","occupancy":true,"trigger_indicator":false,"update":{"state":null},"update_available":null,"voltage":3033}'

Error 2022-06-26 09:56:52Publish 'set' 'motion_sensitivity' to 'Hallway_Aqara_Motion Sensor P1' failed: 'Error: Write 0x54ef441000497017/1 aqaraOpple({"268":{"value":3,"type":32}}, {"sendWhen":"immediate","timeout":10000,"disableResponse":false,"disableRecovery":false,"disableDefaultResponse":true,"direction":0,"srcEndpoint":null,"reservedBits":0,"manufacturerCode":4447,"transactionSequenceNumber":null,"writeUndiv":false}) failed (Data request failed with error: 'MAC transaction expired' (240))'

Thank you!

**Reply by Koenkk** [reply]
This could also be one of the routers messing up. Try to repair the device very close to the coordinator (so it has a direct connection to it), wakeup the device and try to change the detection interval right after that.

**Reply by mdllocalhost** [reply]
That worked! Thank you so much Koenkk!!!

**Reply by javigmsuperka** [reply]
helle guys,

I have a problem trying to change detection_interval.

It only works between 1 and 200 seconds. After the number 200, it just shows the following error:

![image](https://user-images.githubusercontent.com/85594685/198896074-36f4b945-f78f-4b40-a2f4-0140509747f3.png)


Things i did:

1.Wake up the device like documentation says.
2.Re-paired the device directly to the coordinator, same happend. 
3.Reset the devices 5 times, still the same.

**Reply by kkossev** [reply]
200 is the maximum detection interval, according to Aqara docs.

**Reply by javigmsuperka** [reply]
> 200 is the maximum detection interval, according to Aqara docs.

Why does this appear in the zigbee2mqtt documentation?

The minimal value is 2 and the maximum value is 65535. The unit of this value is s.

https://www.zigbee2mqtt.io/devices/RTCGQ14LM.html
